### ISSUE TITLE
induced.subgraph and subgraph.edges don't adjust layout if layout is matrix

### ISSUE THREAD

**Issue by ahmontgo** [igraph/rigraph#219]
I've been plotting multiple subgraphs while attempting to retain layout values, and came across an interesting bug. If g$layout is a matrix, vertex deletion doesn't modify it, which produces odd results. For example:

g <- make_ring(10)
g$layout <- layout_in_circle(g)
plot(induced.subgraph(g,c(1:9)))

The "ghost" node 10 is still plotted, picking up the label of the first vertex to boot!

Workaround is to manually remove the node from layout:
g2 <- induced.subgraph(g,1:9)
g2$layout <- g2$layout[1:9,]
plot(g2)

Similar but slightly more complex manual removal needs to occur when subgraph.edges is used:
g3 <- subgraph.edges(g,1:8)
g3$layout <- g3$layout[V(g)[inc(1:8)],]
plot(g3)

Expected behavior would be for plot() to ignore any "extra" vertices in layout. But depending on internal representation, there may be no way of doing so. In that case, subgraph.edges and induced.subgraph should probably modify any g$layout if g$layout is a matrix.

I'd offer a patch, but my C is very shaky.

Thanks!

**Comment by ntamas** [comment]
Not planned for the time being for sake of backwards compatibility. If you want to adjust the layout when vertices are deleted, you can use vertex attributes named `x` and `y` instead of a graph-level `layout` attribute and this will treat vertex deletion correctly. These attributes are also recognized by `layout_nicely()`.

**Comment by szhorvat** [comment]
I understand that there is no capacity for dealing with this now, but it seems to me that the behaviour is bad enough to be called a bug, which should ideally be tracked.

What the correct behaviour would be is debatable. This issue would provide a space to discuss that.

**Comment by ntamas** [comment]
I'd probably just remove the special handling of the `layout` graph attribute from `layout_nicely()` (which is the only place where this attribute is treated as a layout matrix). Having a vertex-specific quantity stored as a _graph_ attribute is asking for trouble.

**Comment by szhorvat** [comment]
Yes, I agree, and that is of course breaking, so it's probably for 2.0 only. My point is that either there should be a new issue for that or this one should be kept open so we do not forget about this problem that occurs with some usage patterns.

**Comment by schochastics** [comment]
fixed in #1880
